chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -591,11 +550,18 @@ Array Stack(const std::vector<Array>& arrays, int8_t axis) {
     {
         BackwardBuilder bb{"stack", array_refs, out};
         if (BackwardBuilder::Target bt = bb.CreateTarget()) {
-            bt.Define([axis](BackwardContext& bctx) {
+            bt.Define([axis, in_shape = array.shape(), in_dtypes = std::move(in_dtypes)](BackwardContext& bctx) {
                 const Array& gout = *bctx.output_grad();
-                std::vector<Array> gxs = StackGrad(gout, axis);
+                Dtype gout_dtype = gout.dtype();
+                std::vector<Array> gxs = Split(gout, gout.shape()[axis], axis);
```

Comment:
Simpler to reuse `Split` for the backward computation instead of defining a new `StackGrad`.